### PR TITLE
Add countdown timer in the beginning + theme audio and explosion effect

### PR DESCRIPTION
**Pull request by linchen2508**
Add a 3, 2, 1 count down at the beginning to start the race. Creating audio class to play the theme audio and a small explosion sound when hitting the bullseye (in TrackTest scene)

### CODE REVIEW

File: Assets/Scripts/AI/RacingAI.cs
Comment (Aether700) [inline Assets/Scripts/AI/RacingAI.cs]: Instead of hard coding the 3 seconds to wait maybe add the 3 seconds as a variable of the RacerManager and then use a function to get the amount of time to wait. If you need help to add this functionality to the RacerManager let me know .

edit: I saw you have a countdown manager which stores the time to wait. You should query it to know how long to wait instead of hard coding the wait time.

File: Assets/Scripts/AudioManager.cs
Comment (Aether700) [inline Assets/Scripts/AudioManager.cs]: Please remove the unity comments generated

File: Assets/Scripts/CountdownController.cs
Comment (Aether700) [inline Assets/Scripts/CountdownController.cs:11]: small detail but you should use the Text mesh pro text type/component instead of using the default unity text component since the TMP version provides better text display.

File: Assets/Scripts/CountdownController.cs
Comment (Aether700) [inline Assets/Scripts/CountdownController.cs]: Please make your fields private and add the SerializeField attribute so they can be modified in the editor instead of making your fields public.

File: Assets/Scripts/PositionTracker.cs
Comment (Aether700) [inline Assets/Scripts/PositionTracker.cs:81]: Avoid using any function of type "Find" like FindObjectOfType as they are very slow and we can easily find ways around using them. In this case you could make AudioManager a singleton. If you need help to make AudioManager a singleton let me know.

File: Assets/Scripts/PowerUp.cs
Comment (Aether700) [inline Assets/Scripts/PowerUp.cs]: this code should not be here. In the future when you work on a completely new feature base it off of main instead of another branch like the powerup branch

**Review by Aether700 — CHANGES_REQUESTED** [review]
The theme music seems very soft compared to the explosion sound effect. I had raised my volume just to be blasted by the explosion sound. Can you please look into adjusting the volumes?

There are also other changes I would like to request that are explained in comments I made throughout the pr. if you have questions let me know

File: Assets/Scripts/AudioManager.cs
Comment (Aether700) [inline Assets/Scripts/AudioManager.cs:11]: is there a reason why you set up the audio manager to not be destroyed on load?

File: Assets/Scripts/CountdownController.cs
Comment (Aether700) [inline Assets/Scripts/CountdownController.cs]: for your information that is not how you do a singleton pattern. This will work but we cannot modify the count down time through the inspector which isn't great

**Review by Aether700 — COMMENTED** [review]
Much better, I left some comments and I'm waiting for you to answer a question I left in one of the comments before I approve. Otherwise good job.

File: Assets/Scripts/AudioManager.cs
Comment (linchen2508) [reply Assets/Scripts/AudioManager.cs:11]: The youtube tutorial said so

File: Assets/Scripts/CountdownController.cs
Comment (linchen2508) [reply Assets/Scripts/CountdownController.cs]: I didn't do a singleton here, just a getter for the countdown time

File: Assets/Scripts/AudioManager.cs
Comment (Aether700) [reply Assets/Scripts/AudioManager.cs:11]: lol fair enough. In general I would avoid doing so but if the tutorial said so I'm not going to argue

File: Assets/Scripts/CountdownController.cs
Comment (Aether700) [reply Assets/Scripts/CountdownController.cs]: Yeah I saw. A singleton would be better but I won't block the pr for that. if you want to take the time to turn it into a singleton let me know and I can show you.

**Review by Aether700 — DISMISSED** [review]

File: Assets/Scripts/CountdownController.cs
Comment (Aether700) [inline Assets/Scripts/CountdownController.cs]: this field should be private. To allow us to modify it simply add the SerializeField attribute to it and it'll appear in the unity editor

File: Assets/Scripts/CountdownController.cs
Comment (Aether700) [inline Assets/Scripts/CountdownController.cs:15]: that's nice practice but you don't need to do that. In fact if you want to destroy additional instances just make sure to copy the values of the instance so that changes made in the inspector will actually affect the scene otherwise older values from another scene will persist.

Note that this isn't a big issue especially in this specific case

**Review by Aether700 — CHANGES_REQUESTED** [review]
please make the countdownTime a private field. I left more details about it in a comment on the pr

**Review by Aether700 — APPROVED** [review]
Great work

**Pull request merged**
